nixos/modules/installer/sd-card/sd-image-riscv64-visionfive.nix

```diff
@@ -0,0 +1,52 @@
+# To build, use:
+# nix-build nixos -I nixos-config=nixos/modules/installer/sd-card/sd-image-riscv64-visionfive.nix -A config.system.build.sdImage
+{ config, lib, pkgs, ... }:
+
+{
+  imports = [
+    ../../profiles/base.nix
+    ./sd-image.nix
+  ];
+
+  hardware.deviceTree.name = "starfive/jh7100-starfive-visionfive-v1.dtb";
+  systemd.services."serial-getty@hvc0".enable = false;
+  # Must disable when cross-compiling, as the host may not be able to execute a riscv64 program.
+  # Related: #168829
+  services.logrotate.checkConfig = false;
+  environment.systemPackages = with pkgs; [ mtdutils ];
+
+  boot = {
+    consoleLogLevel = lib.mkDefault 7;
+    kernelPackages = pkgs.linuxPackages_visionfive;
+
+    kernelParams = [
+      "console=tty0"
+      "console=ttyS0,115200"
+      "earlycon=sbi"
+    ];
+
+    initrd.kernelModules = [
+      "dw-axi-dmac-platform"
+      "dw_mmc-pltfm"
+      "spi-dw-mmio"
+    ];
+
+    loader = {
+      grub.enable = false;
+      generic-extlinux-compatible.enable = true;
+    };
+  };
+
+  sdImage = {
+    imageName = "${config.sdImage.imageBaseName}-${config.system.nixos.label}-${pkgs.stdenv.hostPlatform.system}-visionfive.img";
+
+    populateFirmwareCommands = ''
+      cp ${pkgs.bootrom-visionfive}/bootrom.bin firmware/opensbi_u-boot_visionfive.bin
```

Comment:
What's the purpose of this?

Reply:
To be able to use this with the hardware, we have the option of flashing from microSD, thus this seems a fitting addition.

Reply:
Specifically, we can use ``load`` to load a binary from the sd card, and use ``sf update`` to overwrite the flash.